### PR TITLE
Some recent repositories settings improvements

### PR DESCRIPTION
**Pull request by pmiossec**
I have selected only the accepted and already reviewed commit of #3769 to see some improvements made for v4.1

Note: Small commits that are easily reviewed one by one.

Configure repositories popup:
* Being able to see the repository selected in the lists
* Being able to do an action on multiple repositories at the same time
* Resize column size to prevent bad display (when destination list was initially empty or changing shortening strategy)
* Display 'Anchored' repositories in bold
* Show full path in a tooltip
* Display unavailable/deleted repositories in red
* Double click to transfer a repository from one list to the other

Repositories list:

* Always display a separator between pinned and all recent repositories

# Screenshots 

### Before

image

### After

image

### Before

image

### After

With separator:

image

## Test methodology 

- Manual

## Test environment(s) 
- Git Extensions 33.33.33
- Build 29515dc7ad189f4959e821a6a266dda8d0b1d288
- Git 2.40.0.windows.1
- Microsoft Windows NT 10.0.22621.0
- .NET 6.0.16
- DPI 96dpi (no scaling)
- Microsoft.WindowsDesktop.App Versions

```
    Microsoft.WindowsDesktop.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

## Merge strategy

I agree that the maintainer "Merge commit" (better) or "Rebase merge" this PR.

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Review by RussKie — APPROVED** [review]
+0

File: GitUI/CommandsDialogs/BrowseDialog/FormRecentReposSettings.cs [hunk 1]
```diff
@@ -153,6 +172,7 @@ private void SetComboWidth()
         private void sortPinnedRepos_CheckedChanged(object sender, EventArgs e)
         {
             RefreshRepos();
+            SetComboWidth();
```
Comment (mstv) [inline GitUI/CommandsDialogs/BrowseDialog/FormRecentReposSettings.cs:175]: ❓ Rather add the call of `SetComboWidth()` to `RefreshRepos()`?
In `comboMinWidthEdit_ValueChanged`, the sequence of calls is interchanged!

File: GitUI/CommandsDialogs/BrowseDialog/FormRecentReposSettings.cs [hunk 2]
```diff
@@ -126,15 +126,34 @@ private void RefreshRepos()
 
             foreach (var repo in pinnedRepos)
             {
-                PinnedLB.Items.Add(new ListViewItem(repo.Caption) { Tag = repo, ToolTipText = repo.Caption });
+                ListViewItem item = GetRepositoryListViewItem(repo, repo.Repo.Anchor == Repository.RepositoryAnchor.Pinned);
+                PinnedLB.Items.Add(item);
```
Comment (mstv) [inline GitUI/CommandsDialogs/BrowseDialog/FormRecentReposSettings.cs:130]: ❔
```suggestion
                PinnedLB.Items.Add(GetRepositoryListViewItem(repo, repo.Repo.Anchor == Repository.RepositoryAnchor.Pinned));
```

File: GitUI/CommandsDialogs/BrowseDialog/FormRecentReposSettings.cs [hunk 3]
```diff
@@ -126,15 +126,34 @@ private void RefreshRepos()
...
             }
 
             foreach (var repo in allRecentRepos)
             {
-                AllRecentLB.Items.Add(new ListViewItem(repo.Caption) { Tag = repo, ToolTipText = repo.Caption });
+                ListViewItem item = GetRepositoryListViewItem(repo, repo.Repo.Anchor == Repository.RepositoryAnchor.AllRecent);
+                AllRecentLB.Items.Add(item);
```
Comment (mstv) [inline GitUI/CommandsDialogs/BrowseDialog/FormRecentReposSettings.cs:136]: ❔ ditto
```suggestion
                AllRecentLB.Items.Add(GetRepositoryListViewItem(repo, repo.Repo.Anchor == Repository.RepositoryAnchor.AllRecent));
```

File: GitUI/CommandsDialogs/BrowseDialog/FormRecentReposSettings.cs [hunk 4]
```diff
@@ -185,32 +205,31 @@ private void Abort_Click(object sender, EventArgs e)
...
+                }
             }
             else
             {
                 e.Cancel = true;
             }
         }
 
-        private bool GetSelectedRepo(object? sender, [NotNullWhen(returnValue: true)] out RecentRepoInfo? repo)
+        private bool GetSelectedRepos(object? sender, [NotNullWhen(returnValue: true)] out List<RecentRepoInfo?> repos)
```
Comment (mstv) [inline GitUI/CommandsDialogs/BrowseDialog/FormRecentReposSettings.cs:224]: ❔  It could output an enumerator. Though performance does not matter here.

**Review by mstv — COMMENTED** [review]
appears sensible to me

**Review by gerhardol — COMMENTED** [review]
+0
Changes seem fine to me but not run and no deep review.
(and pmiossec, I am not innocent of delaying releases myself, but I want the normal update to start so we can get a 4.2 out).

**Review by gerhardol — APPROVED** [review]

**Comment by RussKie** [comment]
@mstv's comments are still unresolved.

**Comment by RussKie** [comment]
@pmiossec could you please resolve the outstanding feedback - action or reject - so that we could merge this. This seems to be the last outstanding item tagged for v4.1. Thanks

**Comment by pmiossec** [comment]
> @pmiossec could you please resolve the outstanding feedback - action or reject - so that we could merge this. This seems to be the last outstanding item tagged for v4.1. Thanks

Done.

File: GitUI/CommandsDialogs/BrowseDialog/FormRecentReposSettings.cs (hunk 4 above)
Comment (pmiossec) [reply GitUI/CommandsDialogs/BrowseDialog/FormRecentReposSettings.cs:221]: I just had a look and I don't think it's possible easily in fact...

File: GitUI/CommandsDialogs/BrowseDialog/FormRecentReposSettings.cs [hunk 5]
```diff
@@ -163,7 +181,6 @@ private void comboMinWidthEdit_ValueChanged(object sender, EventArgs e)
                 AllRecentLB.BeginUpdate();
 
                 SetComboWidth();
-                RefreshRepos();
             }
             finally
             {
```
Comment (mstv) [inline GitUI/CommandsDialogs/BrowseDialog/FormRecentReposSettings.cs:186]:
```
PinnedLB.BeginUpdate();
AllRecentLB.BeginUpdate();
```
seems to be not necessary here any longer.
Though it should be added to `RefreshRepos()`.

Comment (pmiossec) [reply GitUI/CommandsDialogs/BrowseDialog/FormRecentReposSettings.cs]: done (With a little bonus: improvement over the bad UX of the "Combobox minimum width" that I had to use for the 1st time to test this piece of code)

File: GitUI/CommandsDialogs/BrowseDialog/FormRecentReposSettings.cs [hunk 6]
```diff
@@ -35,6 +37,7 @@ private void LoadSettings()
             SetNumericUpDownValue(_NO_TRANSLATE_maxRecentRepositories, AppSettings.MaxPinnedRepositories);
             SetNumericUpDownValue(_NO_TRANSLATE_RecentRepositoriesHistorySize, AppSettings.RecentRepositoriesHistorySize);
 
+            _previousValue = comboMinWidthEdit.Value;
```
Comment (mstv) [inline GitUI/CommandsDialogs/BrowseDialog/FormRecentReposSettings.cs:40]:
```suggestion
            _previousValue = comboMinWidthEdit.Value;

            return;

```

**Review by mstv — APPROVED** [review]

File: GitUI/CommandsDialogs/BrowseDialog/FormRecentReposSettings.cs (hunk 6 above)
Comment (pmiossec) [reply GitUI/CommandsDialogs/BrowseDialog/FormRecentReposSettings.cs:40]: done.

**Pull request merged**